### ISSUE TITLE
Terminal Crash

### ISSUE THREAD

**Issue by Malmik** [offensive-security/kali-nethunter#945]
### Device:

Samsung Callisto (GT-I5510)

### OS version (KitKat/Lollipop/Marshmallow/Nougat):
CyanogenMod 10.2
Android 4.3.1 Jelly Bean

### Built from repo (date and build command) or downloaded from website (links):
I have ported Nethunter via guide on GitHub (yes, i modified and compiled kernel also)

### Output of `cat /proc/version` in adb shell or NetHunter Terminal:
Linux version 2.6.35.7-androidarmv6 (ankur@ankur-PIG31T) (gcc version 4.7 (GCC)
) #1 PREEMPT Fri Mar 21 03:47:36 IST 2014

(I know that version of kernel could be a problem, but really want to port this :) )

## Issue:
When I open NetHunter terminal, it crashes. I know that there were many Issues of this, but strange thing is that when i do:

# su
# bootkali
or
# bootkali_login

there is written:

"mount: mounting /dev/block/stl13 on /data failed
: invalid argument"

and dissapears after half of second/ leaving terminal screen (window 1) with 'Illegal instruction".

And one thing more - when i piped bootkali to text file i got "The Kali chroot is already running! [2J [H"
and after killkali and piping again:
"kernel.shmmax = 268435456
net.ipv4.ip_forward = 1
The Kali chroot has been started.
[2J [H"

My busybox is Stericson v1.26-2

**Comment by jcadduono** [comment]
whoa lol now that is ancient!

i believe that kali rolling chroot has a requirement of at least linux 3.2 for gnu extensions of some kind
it's also likely that armhf instructions used in binaries of the chroot don't exist on your SoC, might be optimized for cortex a7 or a15, not sure what your device has.

**Comment by Malmik** [comment]
Hmmm ok :D 
Is there any way to install these extensions?

Also - I had succesfully working newest Kali distro via linux Deploy, so i think there is a chance to get NetHunter working with it. I can do much, becouse I really want this port cause this is best android device at this moment with sliding QWERTY keyboard that i love! (normally i use my HTC Desire 820n but still like this little phone)

Okey, I must add also that i used a trick to get chroot working on sdcard, because it has around 100 mb of internal storage XD. I just symlinked /data/local/nhsystem to ext4 second partition on my sdcard that i mounted via init.d script. This is very usefull.

I have rooted/unlocked and fixed many phones, and I have basic knowledge in linux and c, so if anyone cleverer than me can support me to do this port, I have some free time and can do as much as I can.
